### PR TITLE
Feature/subcategory url

### PR DESCRIPTION
**Pull request by SDBowen**
**What problem is this solving?**

Adds the option to include a subcategory in the blog URLs, `/blog/category/subcategory`. This can improve user navigation within the blog content and improve SEO.   

Includes a container `div` around the Teaser component, as well as some additional CSS handles
```
postMetaDate
postMetaAuthor
postMetaCategory
```

**How should this be manually tested?**

[workspace with subcategory URLs enabled](https://wpsubcategory--arcaplanet.myvtex.com)
[workspace without subcategory URLs enabled](https://wpcategory--arcaplanet.myvtex.com)

**Screenshots or example usage:**

Screenshot from 2021-06-04 09-27-05

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -314,6 +316,30 @@ Continuing the example from above, any block that shows content from the "defaul
 
 :information_source: _Make sure to follow the format of the example value, including the brackets and escaped double quotes._
 
+### Adding Subcategories to blog URLs
+
+To have subcategories included in your blog URLs (`/blog/category/subcategory`), you will need to update your stres's `routes.json`, as well as the theme's blocks that display category-related URLs.
```
Comment (polishq) [inline docs/README.md:321]:
```suggestion
To have subcategories included in your blog URLs (`/blog/category/subcategory`), you will need to update your store's `routes.json`, as well as the theme's blocks that display category-related URLs.
```

File: docs/README.md [hunk 2]
```diff
@@ -314,6 +316,30 @@ Continuing the example from above, any block that shows content from the "defaul
...
+
+Add a `:subcategoryslug` parameter to the store's blog category route:
+
+```json
+"store.blog-category": {
+	"path": "/blog/category/:categoryslug_id(/:subcategoryslug)"
```
Comment (polishq) [inline docs/README.md:327]: This just occurred to me -- param should probably be `subcategoryslug_id` to enable the site editor to target individual subcategories. Do we know if there's any problem using multiple `_id` params in the same route?

File: react/components/WordpressCategoryBlock.tsx [hunk 3]
```diff
@@ -19,13 +20,14 @@ const CSS_HANDLES = [
 ] as const
 
 const WordpressCategoryBlock: StorefrontFunctionComponent<WPCategoryBlockProps> = ({
-  category,
+  category: catgoryId,
```
Comment (polishq) [inline react/components/WordpressCategoryBlock.tsx:23]:
```suggestion
  category: categoryId,

```

File: react/components/WordpressCategoryBlock.tsx [hunk 4]
```diff
@@ -35,13 +37,30 @@ const WordpressCategoryBlock: StorefrontFunctionComponent<WPCategoryBlockProps>
 }) => {
   const { loading, error, data } = useQuery(CategoryPosts, {
     variables: {
-      category,
+      category: catgoryId,
```
Comment (polishq) [inline react/components/WordpressCategoryBlock.tsx:40]:
```suggestion
      category: categoryId,

```

File: react/components/WordpressCategoryBlock.tsx [hunk 5]
```diff
@@ -102,11 +121,7 @@ const WordpressCategoryBlock: StorefrontFunctionComponent<WPCategoryBlockProps>
           ) : (
             <Link
               page="store.blog-category"
-              params={{
-                categoryslug: data?.wpCategory?.slug,
-                categoryslug_id: data?.wpCategory?.slug,
-                customdomainslug: customDomainSlug,
-              }}
+              params={buildCatelogParams()}
```
Comment (polishq) [inline react/components/WordpressCategoryBlock.tsx:124]:
```suggestion
              params={buildCatalogParams()}

```

File: react/components/WordpressPost.tsx [hunk 6]
```diff
@@ -178,6 +181,20 @@ const WordpressPostInner: FunctionComponent<{
       </div>
     )
 
+  const buildCatelogParams = (category: WPCategory) => {
```
Comment (polishq) [inline react/components/WordpressPost.tsx:184]:
```suggestion
  const buildCatalogParams = (category: WPCategory) => {

```

File: react/components/WordpressPost.tsx [hunk 7]
```diff
@@ -212,22 +229,20 @@ const WordpressPostInner: FunctionComponent<{
...
-            </span>
-          ))}
+          {categories
+            .sort((a, b) => b.parent - a.parent)
+            .map((cat, index) => (
+              <span key={index}>
+                <Link
+                  className={handles.postCategoryLink}
+                  page="store.blog-category"
+                  params={buildCatelogParams(cat)}
```
Comment (polishq) [inline react/components/WordpressPost.tsx:239]:
```suggestion
                  params={buildCatalogParams(cat)}

```

**Review by polishq — CHANGES_REQUESTED** [review]

File: docs/README.md (hunk 2 above)
Comment (SDBowen) [reply docs/README.md:327]: Without the `_id`, is the page expected to not be reachable?

Playing around with the site editor, I can reach a blog subcategory page regardless of having the `_id` identifier. Both versions are acting the same for me.

Comment (polishq) [reply docs/README.md:327]: I think the idea is that the `_id` allows a specific instance of a dynamic route to be targetable. For example if you wanted a banner to appear only on /blog/some-category/some-specific-subcategory . Whereas if the `_id` is left off, the site editor is only able to add a banner that will appear on ALL subcategory pages.

Comment (polishq) [reply docs/README.md:327]: See https://github.com/vtex-apps/wordpress-integration/pull/49

**Review by polishq — CHANGES_REQUESTED** [review]

**Review by polishq — APPROVED** [review]

**Pull request merged**